app/[locale]/academics/helper/courses/CourseDetailModal.tsx

```diff
@@ -1,35 +1,88 @@
+import { useLocale } from 'next-intl';
+import { useReducer, useState } from 'react';
+
+import { deleteCourseAction } from '@/actions/academics';
 import BookmarkIcon from '@/public/image/bookmark_icon.svg';
 
+import { DeleteButton, GrayButton } from '@/components/common/Buttons';
+import LoginVisible from '@/components/common/LoginVisible';
 import ModalFrame from '@/components/modal/ModalFrame';
 
-import { Course } from '@/types/academics';
+import { Course, GRADE } from '@/types/academics';
+import { Language } from '@/types/language';
+
+import { CustomError, handleServerAction } from '@/utils/serverActionError';
+import { errorToast, successToast } from '@/utils/toast';
+
+import CourseEditor from './CourseEditor';
 
 interface CourseDetailModalProps {
-  course: Course;
+  initCourse: Course;
   onClose: () => void;
 }
 
-export default function CourseDetailModal({ course, onClose }: CourseDetailModalProps) {
+export default function CourseDetailModal({ initCourse, onClose }: CourseDetailModalProps) {
+  const [course, setCourse] = useState(initCourse); // 수정 이후 바로 업데이트하기 위함
```

Comment:
```suggestion
  // 수정 이후 페이지 새로고침 없이 모달 내용을 업데이트하기 위해 내부 상태로 관리
  const [course, setCourse] = useState(initCourse); 
```

의도가 이게 맞다면 주석 이렇게 바꾸는거 ㄱㅊ?